File: main/src/main/java/com/crossapplication/main/repository/repo/WeightLogRepository.java
package com.crossapplication.main.repository.repo;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.crossapplication.main.entity.WeightLog;

@Repository
public interface WeightLogRepository extends JpaRepository<WeightLog, Integer>{
    
    List<WeightLog> findByUserIdOrderByDateAsc(int userId);

    List<WeightLog> findByUserIdAndDateBetweenOrderByDateAsc(
            int userId,
            LocalDate startDate,
            LocalDate endDate
    );

    WeightLog findByUserIdAndDate(int userId, LocalDate date);

    WeightLog findTopByUserIdOrderByDateDesc(int userId);

    void deleteByUserId(int userId);
}

File: flutter/flutterfrontend/lib/domain/repositories/weight_log_repository.dart
abstract class WeightLogRepository {
  Future<List<Map<String, dynamic>>> findByUserIdOrderByDateAsc(int userId);
  Future<List<Map<String, dynamic>>> findByUserIdAndDateBetweenOrderByDateAsc(
    int userId,
    DateTime startDate,
    DateTime endDate,
  );
  Future<Map<String, dynamic>?> findByUserIdAndDate(int userId, DateTime date);
  Future<Map<String, dynamic>?> findTopByUserIdOrderByDateDesc(int userId);
  Future<void> deleteByUserId(int userId);
  Future<Map<String, dynamic>?> findById(int id);
  Future<void> save(Map<String, dynamic> weightLog);
  Future<void> update(int id, Map<String, dynamic> weightLog);
  Future<void> delete(int id);
}
